src/components/App.js

```diff
@@ -1,12 +1,40 @@
 import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+import calculateDataObj from '../logic/calculate';
 
-export default function App() {
-  return (
-    <div className="App">
-      <Display />
-      <ButtonPanel />
-    </div>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      obj: {
+        total: null,
+        next: null,
+        operation: null,
+      },
+    };
+  }
+
+  handleClick(btnName) {
+    const { calculate } = calculateDataObj;
+    const obj = this.state;
+    const calculatedObj = calculate(obj, btnName);
+    this.setState({
+      obj: { ...calculatedObj },
+    });
+  }
```

Comment:
- [ ] OPTIONAL - Consider using an arrow function to set state i.e convert handleClick function to an arrow function. That way you wouldn't need to bind handleClick in the constructor.